com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -178,7 +178,7 @@ protected override void Render(ScriptableRenderContext renderContext, Camera[] c
             SetupPerFrameShaderConstants();
 #if ENABLE_VR && ENABLE_XR_MODULE
             SetupXRStates();
-            if(xrSkipRender)
+            if(xrSkipRender && IsGameCamera(cameras[0]))
```

Comment:
Maybe we should consider do this IsGameCamera check on a per camera basis? 
Should we loop through cameras and only do xrSkipRender for game camera?

Reply:
I was wondering that, but I'm also concerned that doesn't make any sense in the current context.

As it originally stands, the `xrSkipRender` var is independent of the camera; it only grabs the active `XRDisplaySubsystems` and checks to see if any of them have a non-zero RenderPass count.

What would we be looking for in said loop? That the array consists of only game cameras? If there is only one non-game camera, what would we expect to happen.

From my testing, that array (`cameras`) is only one element large.

Reply:
IE: would this be appropriate logic:
```
#if ENABLE_VR && ENABLE_XR_MODULE
            SetupXRStates();
            var camerasAreAllGameCameras = true;
            foreach (var camera in cameras)
                camerasAreAllGameCameras &= IsGameCamera(camera);

            if(xrSkipRender && camerasAreAllGameCameras)
                return;
#endif

```

Reply:
In WMR in-editor playmode case with 1 main camera, 2 cameras are provided: 1 sceneview camera and 1 game camera. 
Logic here would render the game camera even if there are 0 XRRenderPass.
In 19.3 + URP 7.x.x, this will cause a editor crash.

Reply:
What do you think about maybe moving xrSkipRender logic to forloop code block below?
```
                if (IsGameCamera(camera)) <-add xrSkipRenderCheck
                {
                    RenderCameraStack(renderContext, camera);
                }
```

Reply:
The problem there is that `xrSkipRender` is defined in an `#if ENABLE_VR && ENABLE_XR_MODULE` block before the `internal void SetupXRStates()` declaration, so it would have to look like this:
```
if (IsGameCamera(camera)
#if ENABLE_VR && ENABLE_XR_MODULE
  && xrSkipRender
#endif
```

That also doesn't early out, like we're doing above. So that's a big change to the current logic.

```